Source/ARTChannel.m

```diff
@@ -51,6 +51,19 @@ - (void)setOptions:(ARTChannelOptions *)options {
 
 - (void)setOptions_nosync:(ARTChannelOptions *)options {
     _options = options;
+    [self recreateDataEncoderWith:options.cipher];
```

Comment:
I worry that here we're changing the behaviour of the `-setOptions:` and `-setOptions_noSync:` methods. Where before they performed their work entirely synchronously, the `dispatch_barrier_async` of `-recreateDataEncoderWith:` makes them asynchronous. This change is, however, not reflected outside of these methods' implementations – if we wanted to make these methods asynchronous, then we'd need to add callbacks to tell the caller when their requested update has completed. That might presumably ultimately have effects on the SDK's public API, so we'd want to tread carefully here.

Reply:
changed here 0b507e371ef454c8b5e6290d1e87afd3db047614